src/edge/tests/test_recombine.py

```diff
@@ -575,6 +575,76 @@ def test_recombines_multiple_times_on_different_fragments(self):
                           't' * 40 + upstream + cassette + downstream
                           + 'c' * 15 + upstream + cassette + downstream + 'c' * 20)
 
+    def test_recombines_multiple_recombinations_each_with_multiple_fragments_on_same_parent_genome(self):
+        upstream = "gagattgtccgcgtttt"
+        front_bs = "catagcgcacaggacgcggag"
+        middle = "cggcacctgtgagccg"
+        back_bs = "taatgaccccgaagcagg"
+        downstream = "gttaaggcgcgaacat"
+        replaced = "aaaaaaaaaaaaaaaaaaaaaa"
+
+        template = ''.join([upstream, front_bs, middle, back_bs, downstream])
+        cassette = ''.join([front_bs, replaced, back_bs])
+
+        f1 = 't' * 20 + template + 'c' * 20 + template + 'c' * 30
+        f2 = 't' * 40 + template + 'c' * 15 + template + 'c' * 20
+
+        arm_len = min(len(front_bs), len(back_bs))
+        g = self.build_genome(False, f1, f2)
+        self.assertEquals(g.fragments.count(), 2)
+
+        # first recombination on g, producing first child c
+
+        c = recombine(g, cassette, arm_len)
+        self.assertEquals(c.fragments.count(), 2)
+        sequences = [f.indexed_fragment().sequence for f in c.fragments.all()]
+        sequences = sorted(sequences, key=lambda s: len(s))
+        self.assertEquals(sequences[0],
+                          't' * 20 + upstream + cassette + downstream
+                          + 'c' * 20 + upstream + cassette + downstream + 'c' * 30)
+        self.assertEquals(sequences[1],
+                          't' * 40 + upstream + cassette + downstream
+                          + 'c' * 15 + upstream + cassette + downstream + 'c' * 20)
+
+        # check original genome sequences are correct
+
+        g2 = Genome.objects.get(pk=g.id)
+        self.assertEquals(g2.fragments.count(), 2)
+        sequences = [f.indexed_fragment().sequence for f in g2.fragments.all()]
+        sequences = sorted(sequences, key=lambda s: len(s))
+        self.assertEquals(sequences[0],
+                          't' * 20 + template + 'c' * 20 + template + 'c' * 30)
```

Comment:
we can just use f1 and f2 here since 
```
f1 = 't' * 20 + template + 'c' * 20 + template + 'c' * 30
f2 = 't' * 40 + template + 'c' * 15 + template + 'c' * 20
```